### PR TITLE
refactor print location decoration for simplicity

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
The current logic is a bit confusing as it's moving a type of `Future[(Option[String], Option[String])]` around and consequently using positional references, which is hard to follow.

Refactor to:
- push print location decoration into its own class
- introduce DecoratedPrintLocation in place of a tuple of optional strings

## How can we measure success?
Easier to understand the print location logic.

## Images
n/a

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala [hunk 1]
```diff
@@ -36,60 +38,17 @@ object StubAPI {
...
-              shortActualPrintLocationDescription = actualPrintLocationDescription._2)
-          )
-        )
-        .copy(
-          longPlannedPrintLocationDescription = plannedPrintLocationDescription._1,
-          shortPlannedPrintLocationDescription = plannedPrintLocationDescription._2
-        )
-      )
+      maybeDecoratedStub <- ApiResponseFt.Async.Right(maybeStub match {
+        case Some(stub) => stubDecorator.withPrintLocationDescriptions(stub).map(Some(_))
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala:47]: `.map(Some(_))` is necessary to satisfy the type. It's not great and help wanted to improve it!

Comment (aug24) [reply common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala:47]:
```suggestion
ApiResponseFt.Async.Right(maybeStub.map(stub => stubDecorator.withPrintLocationDescriptions(stub)))
```

File: common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala [hunk 2]
```diff
@@ -36,60 +38,17 @@ object StubAPI {
...
-              longActualPrintLocationDescription = actualPrintLocationDescription._1,
-              shortActualPrintLocationDescription = actualPrintLocationDescription._2)
-          )
-        )
-        .copy(
-          longPlannedPrintLocationDescription = plannedPrintLocationDescription._1,
-          shortPlannedPrintLocationDescription = plannedPrintLocationDescription._2
-        )
-      )
+      maybeDecoratedStub <- ApiResponseFt.Async.Right(maybeStub match {
```
Comment (aug24) [inline common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala:46]:
```suggestion
      maybeDecoratedStub <- 
```

File: common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala (hunk 1 above)
Comment (akash1810) [reply common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala:47]: 😢  this doesn't satisfy the types.

```scala
ApiResponseFt.Async.Right(maybeStub.map(stub => stubDecorator.withPrintLocationDescriptions(stub)))
```

yields an `Option[Future[Stub]]`, but a `Future[Option[Stub]]` is needed.
